File: solutions/ocaml/hamming/6/hamming.ml
type nucleotide = A | C | G | T ;;

let hamming_distance (a: nucleotide list) (b: nucleotide list) =
  match ( List.length a, List.length b ) with
  | (x, y) when x == y -> List.fold_left (+) 0 ( List.map2 ( fun x y -> if x <> y then 1 else 0 ) a b ) |> Result.ok
  | (0, _) -> Result.error "left strand must not be empty"
  | (_, 0) -> Result.error "right strand must not be empty"
  | _ -> Result.error "left and right strands must be of equal length"

File: solutions/c/hamming/1/hamming.c
#include "hamming.h"

int compute(const char *left, const char *right) {
    if (len(left) == 0 || len(right) == 0) {
        return 0;
    }

    if (len(left) != len(right)) {
        return -1;
    }

    else {
        int distance = 0;
        for (int i = 0; i < len(left); i++) {
            if (left[i] != right[i]) {
                distance++;
            }
        }
        return distance;
    }
}